src/cpp/session/SessionConsoleProcess.cpp

```diff
@@ -251,6 +253,142 @@ void ConsoleProcess::commonInit()
                            module_context::activeSession().id());
 #endif
    }
+   
+   // if RStudio has been configured to use Python,
+   // and the user has opted in to our terminal config
+   // place that copy of Python on the PATH
+   if (prefs::userPrefs().terminalPythonIntegration())
+   {
+      std::string reticulatePython = core::system::getenv("RETICULATE_PYTHON");
+      if (reticulatePython.empty())
+      {
+         Error error = r::exec::RFunction(".rs.inferReticulatePython")
+               .call(&reticulatePython);
+
+         if (error)
+            LOG_ERROR(error);
+
+         if (!reticulatePython.empty())
+         {
+            core::system::setenv(
+                     &*options_.environment,
+                     "RETICULATE_PYTHON",
+                     reticulatePython);
+         }
+      }
+
+      if (!reticulatePython.empty())
+      {
+         auto isBashShell = [&]()
+         {
+            switch (getShellType())
+            {
+            
+            case TerminalShell::ShellType::PosixBash:
+            case TerminalShell::ShellType::GitBash:
+            case TerminalShell::ShellType::WSLBash:
+            {
+               return true;
+            }
+               
+            case TerminalShell::ShellType::CustomShell:
+            {
+               std::string shellPath = options_.shellPath.getAbsolutePath();
+               
+#ifdef _WIN32
+               if (boost::algorithm::ends_with(shellPath, "bash.exe"))
+                  return true;
```

Comment:
Using `bash.exe` to launch WSL is actually considered deprecated, so you should also detect `wsl.exe`.

Reply:
See https://github.com/rstudio/rstudio/issues/6855 for details.

Reply:
Hmm -- the challenge with this, as I understand it, is that `wsl.exe` will actually run whatever shell happens to be configured as the default; e.g. it's the gateway to (probably, but not necessarily) bash. Do you think it's safe for us to still assume it's bash here?

Reply:
I see now you said exactly that below!